src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,279 @@
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+@RestController
+@RequestMapping("/api")
+public class TutorialGroupResource {
+
+    private static final String ENTITY_NAME = "tutorialGroup";
+
+    private final Logger log = LoggerFactory.getLogger(TutorialGroupResource.class);
+
+    private final TutorialGroupService tutorialGroupService;
+
+    private final TutorialGroupRepository tutorialGroupRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final UserRepository userRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    public TutorialGroupResource(AuthorizationCheckService authorizationCheckService, UserRepository userRepository, CourseRepository courseRepository,
+            TutorialGroupService tutorialGroupService, TutorialGroupRepository tutorialGroupRepository) {
+        this.tutorialGroupService = tutorialGroupService;
+        this.courseRepository = courseRepository;
+        this.userRepository = userRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.tutorialGroupRepository = tutorialGroupRepository;
+    }
+
+    /**
+     * GET /tutorial-groups/:tutorialGroupId/title : Returns the title of the tutorial-group with the given id
+     *
+     * @param tutorialGroupId the id of the tutorial group
+     * @return ResponseEntity with status 200 (OK) and with body containing the title of the tutorial group
+     */
+    @GetMapping("/tutorial-groups/{tutorialGroupId}/title")
+    @PreAuthorize("hasRole('USER')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<String> getTitle(@PathVariable Long tutorialGroupId) {
+        final var title = tutorialGroupRepository.getTutorialGroupTitle(tutorialGroupId);
+        return title == null ? ResponseEntity.notFound().build() : ResponseEntity.ok(title);
+    }
+
+    /**
+     * GET courses/:courseId/tutorial-groups: gets the tutorial groups of the specified course.
+     *
+     * @param courseId the id of the course to which the tutorial groups belong to
+     * @return the ResponseEntity with status 200 (OK) and with body containing the tutorial groups of the course
+     */
+    @GetMapping("/courses/{courseId}/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<List<TutorialGroup>> getAllOfCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get all tutorial groups of course with id: {}", courseId);
+        var course = courseRepository.findByIdElseThrow(courseId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
+
+        // ToDo: Optimization Idea: Do not send all registered student information but just the number in a DTO
```

Comment:
I agree. Though in my testing with 2000 users the endpoint actually performed better than expected (<1s), so this might not be that high priority.

Reply:
Will do it in further optimization follow up